hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java

```diff
@@ -102,16 +92,42 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     // client does not need any proto changes.
     OMRequest.Builder omRequest = OMRequest.newBuilder()
         .setUserInfo(getUserInfo())
-        .setUpdateGetS3SecretRequest(updateGetS3SecretRequest)
         .setCmdType(getOmRequest().getCmdType())
         .setClientId(getOmRequest().getClientId());
 
+    // createIfNotExist defaults to true if not specified.
+    boolean createIfNotExist = !s3GetSecretRequest.hasCreateIfNotExist()
+            || s3GetSecretRequest.getCreateIfNotExist();
+
+    // Recompose GetS3SecretRequest just in case createIfNotExist is missing
+    final GetS3SecretRequest newGetS3SecretRequest =
+            GetS3SecretRequest.newBuilder()
+                    .setKerberosID(accessId)
+                    .setCreateIfNotExist(createIfNotExist)
+                    .build();
+    omRequest.setGetS3SecretRequest(newGetS3SecretRequest);
+
+    // When createIfNotExist is true, pass UpdateGetS3SecretRequest;
+    // otherwise, just use GetS3SecretRequest.
+    if (createIfNotExist) {
+      // Generate secret. Used only when doesn't the accessId entry doesn't
+      //  exist in DB, discarded otherwise.
```

Comment:
nitpick: fix comment

Reply:
Hmm this comment is accurate, actually. It might have been a bit misleading. I am pushing an update.

Note this `S3GetSecretRequest` is not a new class. This is shared between the existing `ozone s3 getsecret`, and the new `ozone tenant s3 getsecret`.